transport/tchannel/outbound.go

```diff
@@ -97,11 +97,11 @@ func (o *Outbound) Call(ctx context.Context, req *transport.Request) (*transport
 func (p *tchannelPeer) Call(ctx context.Context, req *transport.Request) (*transport.Response, error) {
 	root := p.transport.ch.RootPeers()
 	tp := root.GetOrAdd(p.HostPort())
-	return callWithPeer(ctx, req, tp)
+	return callWithPeer(ctx, req, tp, p.transport.originalHeaders)
 }
 
 // callWithPeer sends a request with the chosen peer.
-func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Peer) (*transport.Response, error) {
+func callWithPeer(ctx context.Context, req *transport.Request, peer *tchannel.Peer, originalHeader bool) (*transport.Response, error) {
```

Comment:
boolean trap. begging for an enum, like `type headerCase` with values 0 `canonicalizedHeaderCase` vs 1 `originalHeaderCase`.